File: frontend/src/domain/fundstelle.ts
export interface Periodikum {
  readonly id: string
  abbreviation: string
  title: string
  subtitle?: string
  citationStyle?: string
}

export interface Fundstelle {
  readonly id: string
  zitatstelle: string
  periodikum?: Periodikum
  ambiguousPeriodikum?: string
}

File: backend/src/main/java/de/bund/digitalservice/ris/adm_vwv/application/Fundstelle.java
package de.bund.digitalservice.ris.adm_vwv.application;

import java.util.UUID;

/**
 * A reference to a legal periodical (Fundstelle).
 *
 * @param id                  The id of the reference.
 * @param zitatstelle         The citation
 * @param periodikum          The periodikum, where reference was found.
 *                            If multiple periodika were found with same
 *                            abbreviation then this field is set to null and
 *                            ambiguous periodikum is set instead.
 * @param ambiguousPeriodikum If set, periodikum is not unique
 */
public record Fundstelle(
  UUID id,
  String zitatstelle,
  LegalPeriodical periodikum,
  String ambiguousPeriodikum
) {}
